### PR TITLE
fix:  raise EnvoyCommunicationError for httpx ConnectError and TimeoutException exceptions during Envoy.update

### PR DESCRIPTION
**Pull request by catsmanac**
When a httpx connection or timeout error occurs during update, it results in a stack dump in HA log file.

This PR adds exception catch for these and raising EnvoyCommunicationError.

### CODE REVIEW

File: src/pyenphase/envoy.py [hunk 1]
```diff
@@ -441,6 +441,8 @@ async def update(self) -> EnvoyData:
                 await updater.update(data)
             except EndOfStream as err:
                 raise EnvoyCommunicationError("EndOfStream at update") from err
+            except (httpx.ConnectError, httpx.TimeoutException) as err:
+                raise EnvoyCommunicationError(f"{err}") from err
```
Comment (bdraco) [inline src/pyenphase/envoy.py:445]: Maybe we want a separate exception for timeouts?

Comment (bdraco) [inline src/pyenphase/envoy.py:445]:
```suggestion
                raise EnvoyCommunicationError(str(err)) from err
```

**Review by bdraco — APPROVED** [review]

File: src/pyenphase/envoy.py (hunk 1 above)
Comment (catsmanac) [reply src/pyenphase/envoy.py]: > Maybe we want a separate exception for timeouts?

probably best as the httpx.TimeoutException does not return any description.

Comment (catsmanac) [reply src/pyenphase/envoy.py]: Split them and changed httpx.ConnectError to httpx.NetworkError to catch more general httpx network errors.

**Review by bdraco — APPROVED** [review]

**Pull request merged**